src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -110,44 +121,126 @@ protected override void Dispose(bool disposing)
 
         #region Private methods
 
-        private void StartRequest(RequestCompletionSource state)
+        private async void StartRequest(RequestCompletionSource state)
```

Comment:
@davidsh , can you also comment on the buffering requirement?
@stephentoub, note that the read may never happen. If the response body is large and we use a variant of MemoryStream. lot of memory gets used up. We would then have to look at options like saving on file system

I have now removed the usage of StartNew and simplified the task completion logic